src/core/gui/XournalView.cpp

```diff
@@ -139,6 +139,45 @@ auto XournalView::onKeyPressEvent(GdkEventKey* event) -> bool {
         }
     }
 
+    if (auto* tool = getControl()->getWindow()->getPdfToolbox(); tool->hasSelection()) {
+        if (event->keyval == GDK_KEY_c && event->state & GDK_CONTROL_MASK) {
+            // Shortcut to get selected PDF text.
+            tool->copyTextToClipboard();
+            return true;
+        }
+    }
+
+    if (auto* selection = getSelection(); selection) {
+        if (event->keyval == GDK_KEY_Escape) {
+            clearSelection();
+            return true;
+        }
+
+        int d = 3;
+        if (event->state & GDK_MOD1_MASK) {
+            d = 1;
+        } else if (event->state & GDK_SHIFT_MASK) {
+            d = 20;
+        }
+
+        int xdir = 0;
+        int ydir = 0;
+        if (event->keyval == GDK_KEY_Left) {
```

Comment:
The [GTK4 migration guide](https://docs.gtk.org/gtk4/migrating-3to4.html#stop-using-direct-access-to-gdkevent-structs) suggests to avoid directly accessing member variables of GdkEvent.
They say gdk3 already has some accessors, but in this case I'm not sure they match the gdk4 ones.
Anyway, if you feel like it, you can try and adapt those accesses to limit the gtk4 porting effort down the road. I don't how much of that is actually possible, so feel free to ignore this comment.

Reply:
Right, I keep forgetting about that. There is `gdk_event_get_keyval` (like already used e.g. in GeometryToolInputHandler)

Reply:
Yes, but gdk4 only has gdk_key_event_get_keyval taking a GdkKeyEvent (but it still has gdk_event_get_modifier_state)